ssl/src/main/java/org/wildfly/security/ssl/X509RevocationTrustManager.java

```diff
@@ -394,12 +394,8 @@ private void checkCertPathLength(X509Certificate currCert) throws CertPathValida
             X500Principal issuer = currCert.getIssuerX500Principal();
 
             int pathLenConstraint = -1;
-            if (currCert.getVersion() < 3) {    // version 1 or version 2
-                if (i == 1) {
-                    if (subject.equals(issuer)) {
-                        pathLenConstraint = Integer.MAX_VALUE;
-                    }
-                }
+            if (currCert.getVersion() < 3 && i == 1 && subject.equals(issuer)) {    // version 1 or version 2
+                pathLenConstraint = Integer.MAX_VALUE;
```

Comment:
Hi @gabrielpadilh4 
Thank you for your PR. 
We need to be a bit careful about this. The outer if statement has an else statement, so whenever that check is false, the code would jump to the else statement. However, with the new code here, even if the outer statement `currCert.getVersion() < 3` is true, as long at least one of ` i == 1 ` or `subject.equals(issuer)` is false, the code would jump to the else statement since this is a 3 way AND.
Please update your code to restore it to the old logic.
If you have any questions, please feel free to ping me.

Reply:
Hello @PrarthonaPaul so it should be like this ?
```
            if (currCert.getVersion() < 3) {    // version 1 or version 2
                if (i == 1 && subject.equals(issuer)) {
                        pathLenConstraint = Integer.MAX_VALUE;
                }
```

I think this make more sense since it only enters in the first if `currCert.getVersion() < 3`. I believe that we maintain the logic as the only way to set the `pathLenConstraints` is if `i==1 AND subject.equals(issuer)`.

What do you think ? Thank you.

Reply:
I already did the change, could you please review ?

Reply:
Yes, thats correct! 
Thank you for making the changes!